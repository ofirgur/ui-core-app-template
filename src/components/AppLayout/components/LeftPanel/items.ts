import routes from 'router/routes';

const { manageproducts, createajob, manageavatars, managegarments, status } = routes;
export const LIST_ITEMS = [
  manageproducts,
  createajob,
  manageavatars,
  managegarments,
  status,
];