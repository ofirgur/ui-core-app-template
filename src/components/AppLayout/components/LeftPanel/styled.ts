import styled from '@emotion/styled';
import { blue, grey } from '@mui/material/colors';

export const StyledLeftPanel = styled.div``;

export const StyledLogo = styled.div`
  height: 80px;
  line-height: 80px;
  font-size: 25px;
  font-weight: 600;
  padding: 0 10px;
`;

export const StyledList = styled.ul`
  width: 200px;
`;

type StyledListItemType = {
  active: boolean;
};

export const StyledListItem = styled.li<StyledListItemType>`
  display: flex;
  align-items: center;
  height: 50px;
  cursor: pointer;
  padding: 10px;
  font-size: 18px;
  &:hover {
    background-color: ${blue[300]};
    font-weight: bold;
  }

  ${(p) =>
    p.active &&
    `
    background: ${grey[100]};
    color: #000;
  `}
`;
