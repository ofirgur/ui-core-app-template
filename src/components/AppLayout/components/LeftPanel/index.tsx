import React from 'react';
import { useNavigate } from 'react-router-dom';
import STRINGS from '../../../../utils/strings';
import { useCurrentRoute } from 'router/hooks';
import { LIST_ITEMS } from './items';
import {
  StyledLeftPanel,
  StyledLogo,
  StyledList,
  StyledListItem,
} from './styled';

const LeftPanel = () => {
  const currentRoute = useCurrentRoute();
  const navigate = useNavigate();
  const handleNavigate = (pathname: string) => () => {
    navigate(pathname);
  };

  const isActiveRoute = (pathname: string) => {
    if (!currentRoute) return false;
    return currentRoute.pathname.includes(pathname);
  };

  return (
    <StyledLeftPanel>
      <StyledLogo>{STRINGS.Logo}</StyledLogo>
      <StyledList>
        {LIST_ITEMS.map((listItem) => {
          return (
            <StyledListItem
              key={listItem.pathname}
              onClick={handleNavigate(listItem.pathname)}
              active={isActiveRoute(listItem.pathname)}
            >
              {listItem.title}
            </StyledListItem>
          );
        })}
      </StyledList>
    </StyledLeftPanel>
  );
};

export default LeftPanel;
