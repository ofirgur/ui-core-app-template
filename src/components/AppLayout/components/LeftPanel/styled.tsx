import styled from 'styled-components';
import variables from 'utils/variables';

export const LeftPanelStyled = styled.ul`
  width: ${variables.layout.leftPanel.width}px;
  margin-top: 70px;
`;

export const ListItemStyled = styled.li`
  display: flex;
  align-items: center;
  height: 50px;
  cursor: pointer;
  padding: 10px;
  font-size: 20px;
  &:hover {
    background-color: #262626;
  }
`;
