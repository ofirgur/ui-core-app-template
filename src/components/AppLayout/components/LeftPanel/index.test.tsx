import * as React from 'react';
import { render } from '@testing-library/react';
import LeftPanel from '.';

describe('LeftPanel', () => {
  it('should render correctly', () => {
    const { container } = render(<LeftPanel />);
    expect(container).toMatchSnapshot();
  });
});
