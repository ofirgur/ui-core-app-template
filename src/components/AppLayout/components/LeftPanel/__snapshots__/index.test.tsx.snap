// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftPanel should render correctly 1`] = `
<div>
  <div
    class="sc-bczRLJ hDKnho"
  >
    Left Panel
  </div>
</div>
`;
